File: src/server/sys_handler.cpp
#include "kallisto/server/sys_handler.hpp"
#include <ctime>

namespace kallisto {
namespace server {

void SysHandler::handleRequest(HttpHandler& handler, HttpHandler::Connection& conn, const HttpHandler::HttpRequest& req) {
    if (req.path == "/v1/sys/mounts") {
        if (req.method == "GET") {
            handleMounts(handler, conn);
        } else {
            handler.sendError(conn, 405, "Method Not Allowed");
        }
    } else if (req.path == "/v1/sys/health") {
        if (req.method == "GET") {
            handleHealth(handler, conn);
        } else {
            handler.sendError(conn, 405, "Method Not Allowed");
        }
    } else if (req.path == "/v1/sys/seal-status") {
        if (req.method == "GET") {
            handleSealStatus(handler, conn);
        } else {
            handler.sendError(conn, 405, "Method Not Allowed");
        }
    } else {
        handler.sendError(conn, 404, "Not Found");
    }
}

void SysHandler::handleMounts(HttpHandler& handler, HttpHandler::Connection& conn) {
    std::string json = R"({
  "request_id": "c1a2b3c4-d5e6-f7a8-b9c0-112233445566",
  "renewable": false,
  "lease_duration": 0,
  "data": {
    "secret/": {
      "uuid": "4db6db99-197e-128a-78f9-901ab23cd45e",
      "type": "kv",
      "description": "key/value secret storage",
      "config": {
        "default_lease_ttl": 0,
        "max_lease_ttl": 0,
        "force_no_cache": false
      },
      "options": {
        "version": "2"
      },
      "accessor": "kv_accessor_123456"
    }
  }
})";
    handler.sendResponse(conn, 200, "application/json", json);
}

void SysHandler::handleHealth(HttpHandler& handler, HttpHandler::Connection& conn) {
    std::string json = R"({
  "initialized": true,
  "sealed": false,
  "standby": false,
  "performance_standby": false,
  "replication_performance_class": "primary",
  "replication_dr_class": "primary",
  "server_time_utc": )" + std::to_string(std::time(nullptr)) + R"(,
  "version": "1.15.0",
  "cluster_name": "kallisto-cluster-default",
  "cluster_id": "cluster_99887766"
})";
    handler.sendResponse(conn, 200, "application/json", json);
}

void SysHandler::handleSealStatus(HttpHandler& handler, HttpHandler::Connection& conn) {
    std::string json = R"({
  "type": "shamir",
  "initialized": true,
  "sealed": false,
  "t": 1,
  "n": 1,
  "progress": 0,
  "nonce": "",
  "version": "1.15.0",
  "cluster_name": "kallisto-cluster-default",
  "cluster_id": "cluster_99887766"
})";
    handler.sendResponse(conn, 200, "application/json", json);
}

} // namespace server
} // namespace kallisto

File: src/server/sys_handler.rs
use axum::{routing::get, Json, Router};
use serde_json::{json, Value};

/// Mock system endpoints representing Vault's sys/ api
pub fn router<S: Clone + Send + Sync + 'static>() -> Router<S> {
    Router::new()
        .route("/health", get(health_check))
        .route("/mounts", get(get_mounts))
        .route("/seal-status", get(seal_status))
        .route("/internal/ui/mounts/*path", get(get_internal_ui_mounts))
}

async fn health_check() -> Json<Value> {
    Json(json!({
        "initialized": true,
        "sealed": false,
        "standby": false,
        "performance_standby": false,
        "replication_performance_mode": "disabled",
        "replication_dr_mode": "disabled",
        "server_time_utc": 0,
        "version": "1.13.0",
        "cluster_name": "kallisto-cluster",
        "cluster_id": "mock-cluster-id"
    }))
}

async fn get_mounts() -> Json<Value> {
    Json(json!({
        "secret/": {
            "accessor": "kv_mock",
            "config": {
                "default_lease_ttl": 0,
                "force_no_cache": false,
                "max_lease_ttl": 0
            },
            "description": "key/value secret storage",
            "local": false,
            "options": {
                "version": "2"
            },
            "seal_wrap": false,
            "type": "kv",
            "uuid": "mock-uuid"
        }
    }))
}

async fn get_internal_ui_mounts() -> Json<Value> {
    Json(json!({
        "options": {
            "version": "2"
        },
        "path": "secret/",
        "type": "kv"
    }))
}

async fn seal_status() -> Json<Value> {
    Json(json!({
        "type": "shamir",
        "initialized": true,
        "sealed": false,
        "t": 1,
        "n": 1,
        "progress": 0,
        "nonce": "",
        "version": "1.13.0",
        "migration": false,
        "cluster_name": "kallisto-cluster",
        "cluster_id": "mock-cluster-id",
        "recovery_seal": false
    }))
}

#[cfg(test)]
mod tests {
    use super::*;
    use axum::{body::Body, http::Request};
    use tower::ServiceExt;

    #[tokio::test]
    async fn test_health_check() {
        let app = router();
        let response = app
            .oneshot(Request::builder().uri("/health").body(Body::empty()).unwrap())
            .await
            .unwrap();

        assert_eq!(response.status(), 200);
        let body_bytes = axum::body::to_bytes(response.into_body(), usize::MAX).await.unwrap();
        let body: Value = serde_json::from_slice(&body_bytes).unwrap();
        
        assert_eq!(body["initialized"], true);
        assert_eq!(body["sealed"], false);
        assert_eq!(body["standby"], false);
    }

    #[tokio::test]
    async fn test_mounts() {
        let app = router();
        let response = app
            .oneshot(Request::builder().uri("/mounts").body(Body::empty()).unwrap())
            .await
            .unwrap();

        assert_eq!(response.status(), 200);
        let body_bytes = axum::body::to_bytes(response.into_body(), usize::MAX).await.unwrap();
        let body: Value = serde_json::from_slice(&body_bytes).unwrap();
        
        assert_eq!(body["secret/"]["type"], "kv");
    }

    #[tokio::test]
    async fn test_seal_status() {
        let app = router();
        let response = app
            .oneshot(Request::builder().uri("/seal-status").body(Body::empty()).unwrap())
            .await
            .unwrap();

        assert_eq!(response.status(), 200);
        let body_bytes = axum::body::to_bytes(response.into_body(), usize::MAX).await.unwrap();
        let body: Value = serde_json::from_slice(&body_bytes).unwrap();
        
        assert_eq!(body["type"], "shamir");
        assert_eq!(body["sealed"], false);
    }
}
